app/models/user.rb

```diff
@@ -11,18 +12,20 @@ class User < ActiveRecord::Base
 
   class << self
     def find_for_twitter_oauth(auth)
-      user = User.find_by_twitter_oauth(auth["credentials"]["token"])
+      user = User.where(uid: auth.uid, provider:auth.provider).first
 
       if user
         user
       else
         User.create(
-          name: auth["info"]["name"],
-          twitter_handle: auth["info"]["nickname"],
-          twitter_description: auth["info"]["description"],
-          website: (auth["info"]["urls"]["Website"] rescue nil),
-          twitter_oauth: auth["credentials"]["token"],
-          image: auth["info"]["image"]
+            uid: auth["uid"],
+            provider: auth["provider"],
+            name: auth["info"]["name"],
+            twitter_handle: auth["info"]["nickname"],
+            twitter_description: auth["info"]["description"],
+            website: (auth["info"]["urls"]["Website"] rescue nil),
+            twitter_oauth: auth["credentials"]["token"],
+            image: auth["info"]["image"]
```

Comment:
Indentation correction -should be two spaces - looks like its's four spaces inside the create block